src/main/java/org/opentripplanner/middleware/utils/ConfigUtils.java

```diff
@@ -53,6 +58,38 @@ public static void loadConfig(String[] args) throws IOException {
             envConfigStream = new FileInputStream(new File(args[0]));
         }
         envConfig = yamlMapper.readTree(envConfigStream);
+        validateConfig();
+    }
+
+    /**
+     * Validate the environment configuration against the environment configuration schema. If the config is not available,
+     * does not match the schema or an exception is thrown, exit the application.
+     */
+    private static void validateConfig() {
+        if ("false".equals(getConfigPropertyAsText("VALIDATE_ENVIRONMENT_CONFIG", "true"))) {
+            LOG.warn("Environment configuration schema validation disabled.");
+            return;
+        }
+        try {
+            if (envConfig == null) {
+                throw new IllegalArgumentException("Environment configuration not available to validate!");
+            }
+            FileInputStream envSchemaStream = new FileInputStream(new File(DEFAULT_ENV_SCHEMA));
+            JsonNode envSchema = yamlMapper.readTree(envSchemaStream);
+            if (envSchema == null) {
+                throw new IllegalArgumentException("Environment configuration schema not available.");
+            }
+            ProcessingReport report = JsonSchemaFactory
+                    .byDefault()
+                    .getValidator()
+                    .validate(envSchema, envConfig);
+            if (!report.isSuccess()) {
+                throw new IllegalArgumentException(report.toString());
```

Comment:
If a parameter is missing in `env.yml`, Is it possible to not display the list of required parameters?

Reply:
@binh-dam-ibigroup I'm sure it is possble. Could you explain why you think it should be removed please?

Reply:
That would simplify the error message IMO.

Reply:
ok, unless you feel really strongly about this I'm going to leave it as is. The change would require constucting my own report instead of using the default one.

Reply:
I agree that a custom report is not worth building at this time.